presto-hive/src/test/java/com/facebook/presto/hive/TestHiveDistributedQueries.java

```diff
@@ -116,5 +120,31 @@ public void testTrackMaterializedCTEs()
         checkCTEInfo(explain, "tbl2", 1, false, true);
     }
 
+    @Test
+    public void testExplainAnalyzeMaterializedCTEs()
+    {
+        Session materializedSession = Session.builder(getSession())
+                .setSystemProperty(VERBOSE_OPTIMIZER_INFO_ENABLED, "true")
+                .setSystemProperty(CTE_MATERIALIZATION_STRATEGY, "ALL")
+                .setSystemProperty(CTE_PARTITIONING_PROVIDER_CATALOG, "hive")
+                .build();
+
+        JsonRenderer renderer = new JsonRenderer(getQueryRunner().getMetadata().getFunctionAndTypeManager());
+
+        List<Map<PlanFragmentId, JsonRenderer.JsonPlan>> fragmentsList = renderer.deserialize((String) computeActual(materializedSession, "EXPLAIN ANALYZE (format JSON) SELECT * FROM orders").getOnlyValue());
```

Comment:
So I see you copied some queries from the other test case. We actually need a new query which executes with some CTE materialization.

Instead of all these, let's just have one query:

```sql
EXPLAIN ANALYZE with x as (SELECT * from orders WHERE totalprice < 1000) SELECT avg(totalprice) FROM x;
```

This should generate a query with 1 CTE which splits the execution into two stages and should fail without your changes

Reply:
Changed the query